modules/gateway/peers.go

```diff
@@ -52,8 +53,23 @@ func (g *Gateway) addPeer(p *peer) {
 	go g.listenPeer(p)
 }
 
+// randomPeer returns a random peer from the gateway's peer list.
+func (g *Gateway) randomPeer() (modules.NetAddress, error) {
+	if len(g.peers) > 0 {
+		r := rand.Intn(len(g.peers))
```

Comment:
You can ask for 30 random peers, do that enough times and you'll start to get an idea through the randomness. People have done really impressive things by manipulating less than perfectly-random numbers, and I don't want them doing equally impressive things to us.

Additionally, maps are not psuedo-random. They are very low entropy, and usually quite predictable.

Is it that bad to use crypto/rand instead of math/rand?